kernel/src/hil/touch.rs

```diff
@@ -0,0 +1,117 @@
+//! Interface for touch input devices
+
+use crate::ReturnCode;
+
+/// Touch Event Status
+#[derive(Debug, Copy, Clone)]
+pub enum TouchStatus {
+    Pressed,
+    Released,
+}
+
+/// Gesture event
+#[derive(Debug, Copy, Clone)]
+pub enum GestureEvent {
+    SwipeUp,
+    SwipeDown,
+    SwipeLeft,
+    SwipeRight,
+    ZoomIn,
+    ZoomOut,
+}
+
+/// A single touch event's data
+#[derive(Copy, Clone)]
+pub struct TouchEvent {
+    pub status: TouchStatus,
+    /// touch (x, y) position
+    pub x: usize,
+    pub y: usize,
+
+    /// Numeric ID assigned to this touch. This ID allows the client to
+    /// to match different `TouchEvent`s to the same physical touch.
+    ///
+    /// The driver must assign a unique ID to each touch.
+    pub id: usize,
+
+    /// Optional scaled value for the size of the touch. A larger value
+    /// corresponds to a "fatter" touch. The size values range from 0
+    /// to 65535.
+    ///
+    /// If a touchscreen does not provide information about the size of the touch,
+    /// this must be set to `None`.
+    pub size: Option<usize>,
```

Comment:
If the range is 0..65535, should this be a `u16` rather than a `usize`?

Reply:
Separately: I'm mildly concerned that we're encoding something specific to one touch driver in a general-purpose HIL here. I think this is okay for today; we have changed HILs before, and this is a reasonable enough range for drivers to adapt to.

Reply:
I agree with u16, I don't think there will ever be a screen that large